packages/camera/camera_avfoundation/ios/Classes/FLTCam.m

```diff
@@ -672,6 +672,10 @@ - (void)captureOutput:(AVCaptureOutput *)output
     if (_videoWriter.status != AVAssetWriterStatusWriting) {
       [_videoWriter startWriting];
       [_videoWriter startSessionAtSourceTime:currentSampleTime];
+      // fix sample times not being numeric after pause/resume
+      // https://github.com/flutter/flutter/issues/132014
+      _lastVideoSampleTime = currentSampleTime;
+      _lastAudioSampleTime = currentSampleTime;
```

Comment:
are these already set when we pause?

Reply:
The problem was happening because if pause and resume happened before the first sample buffer arrived then they would not be properly initialised when first used, they would have zero timescale/denominator and `CMTimeAdd` and `CMTimeSubtract` propagate that "non-numeric" value through `_videoTimeOffset` into `nextSampleTime` (or if this is not the first recording then they will contain some old timestamp from previous recording).

Reply:
Interesting. It isn't clear from the code. Can you add this reasoning as a comment in code?